src/Chapter4.hs

```diff
@@ -472,10 +476,12 @@ Implement the Applicative instance for our 'Secret' data type from before.
 -}
 instance Applicative (Secret e) where
     pure :: a -> Secret e a
-    pure = error "pure Secret: Not implemented!"
+    pure = Reward
 
     (<*>) :: Secret e (a -> b) -> Secret e a -> Secret e b
-    (<*>) = error "(<*>) Secret: Not implemented!"
+    (<*>) (Reward f) (Reward a) = Reward $ f a
+    (<*>) (Trap e) _ = Trap e
+    (<*>) _ (Trap e) = Trap e
```

Comment:
Note that the first and the last cases look very alike the `Functor` instance 🙂 Alternatively, you could reuse it here, and have 2 cases instead of 3 👍🏼